drivers/soundwire/master.c

```diff
@@ -58,6 +58,12 @@ struct sdw_master_device *sdw_md_add(struct sdw_md_driver *driver,
 	}
 
 	md->dev.driver = &driver->driver;
+	ret = device_bind_driver(&md->dev);
```

Comment:
don't we need a matching device_release_driver() ?

Reply:
@bardliao typically device_add should take care of binding the driver isnt it? I may be wrong. But is the problem simply that we should set md->dev.driver before we call device_resgiter() in sdw_md_add?

Reply:
> @bardliao typically device_add should take care of binding the driver isnt it? I may be wrong. But is the problem simply that we should set md->dev.driver before we call device_resgiter() in sdw_md_add?

@ranj063 You are right. Actually, I tried it before. But we didn't call driver_register() at that time so it result in system crash. Now moving md->dev.driver before device_resgiter() work fine.